lms/djangoapps/teams/tests/test_api.py

```diff
@@ -76,3 +79,163 @@ def test_public_discussion_visible_by_user(self):
         self.assertTrue(teams_api.discussion_visible_by_user('DO_NOT_EXISTS', self.user3))
+
+
+@ddt.ddt
+class TeamAccessTests(SharedModuleStoreTestCase):
+    """
+    The set of tests for API endpoints related to access of a team based on the users
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TeamAccessTests, cls).setUpClass()
+        cls.user_audit = UserFactory.create(username='user_audit')
+        cls.user_staff = UserFactory.create(username='user_staff')
+        cls.user_masters = UserFactory.create(username='user_masters')
+        cls.user_unenrolled = UserFactory.create(username='user_unenrolled')
+        cls.users = {
+            'user_audit': cls.user_audit,
+            'user_staff': cls.user_staff,
+            'user_masters': cls.user_masters,
+            'user_unenrolled': cls.user_unenrolled,
+        }
+
+        for user in (cls.user_audit, cls.user_staff):
+            CourseEnrollmentFactory.create(user=user, course_id=COURSE_KEY1)
+        CourseEnrollmentFactory.create(user=cls.user_masters, course_id=COURSE_KEY1, mode=CourseMode.MASTERS)
+
+        CourseStaffRole(COURSE_KEY1).add_users(cls.user_staff)
+
+        cls.topic_id = 'RANDOM TOPIC'
+        cls.team_unprotected_1 = CourseTeamFactory(
+            course_id=COURSE_KEY1,
+            topic_id=cls.topic_id,
+            team_id='team_unprotected_1'
+        )
+        cls.team_protected_1 = CourseTeamFactory(
+            course_id=COURSE_KEY1,
+            team_id='team_protected_1',
+            topic_id=cls.topic_id,
+            organization_protected=True
+        )
+        cls.team_unprotected_2 = CourseTeamFactory(
+            course_id=COURSE_KEY1,
+            topic_id=cls.topic_id,
+            team_id='team_unprotected_2'
+        )
+        cls.team_unprotected_3 = CourseTeamFactory(
+            course_id=COURSE_KEY1,
+            topic_id=cls.topic_id,
+            team_id='team_unprotected_3'
+        )
+        cls.team_protected_2 = CourseTeamFactory(
+            course_id=COURSE_KEY1,
+            team_id='team_protected_2',
+            topic_id=cls.topic_id,
+            organization_protected=True
+        )
+
+    @ddt.data(
+        ('user_audit', True),
+        ('user_masters', True),
+        ('user_staff', True),
+        ('user_unenrolled', False),
+    )
+    @ddt.unpack
+    def test_has_team_api_access(self, username, expected_have_access):
+        user = self.users[username]
+        self.assertEqual(
+            expected_have_access,
+            teams_api.has_team_api_access(user, COURSE_KEY1)
+        )
+
+    @ddt.data(
+        ('user_audit', teams_api.ORGANIZATION_UNPROTECTED),
+        ('user_masters', teams_api.ORGANIZATION_PROTECTED),
+        ('user_staff', teams_api.ORGANIZATION_PROTECTION_ADMIN),
+        ('user_unenrolled', None),
+    )
+    @ddt.unpack
+    def user_organization_proctection_status(self, username, expected_protection_status):
```

Comment:
s/proctection/protection

also, I'm not sure that this test function will run if its name doesn't start with `test_`